src/Symfony/Component/Validator/Constraints/UlidValidator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * Validates whether the value is a valid ULID (Universally Unique Lexicographically Sortable Identifier).
+ * Cf https://github.com/ulid/spec for ULID specifications.
+ *
+ * @author Laurent Clouet <laurent35240@gmail.com>
+ */
+class UlidValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Ulid) {
+            throw new UnexpectedTypeException($constraint, Ulid::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        $value = (string) $value;
+
+        if (26 > \strlen($value)) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setCode(Ulid::TOO_SHORT_ERROR)
+                ->addViolation();
+        }
+
+        if (26 < \strlen($value)) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setCode(Ulid::TOO_LONG_ERROR)
+                ->addViolation();
+        }
+
+        if (\strlen($value) !== strspn($value, '0123456789ABCDEFGHJKMNPQRSTVWXYZabcdefghjkmnpqrstvwxyz')) {
```

Comment:
`26 !== strspn(`[...]

Reply:
I think it makes more sense to keep `\strlen($value)` otherwise a too short ULID but with only valid characters will have violations about being too short AND having invalid characters.
The violation about invalid characters would not be true in this case.

Let me know what you think about it